### PR TITLE
Use user.is_authenticated for Django 2.0 compatibility.

### PR DESCRIPTION
**Pull request by Anteru**
Fix for #120.

### CODE REVIEW

**Comment by jcassee** [comment]
Thanks, for the PR, @Anteru, but django-ananytical still supports Django 1.8 (and thus Python 2). Could you update the PR 1) include Django 2.0 to the test set, and 2) fix backward-compatibility with 1.8?

**Comment by jcassee** [comment]
LGTM, @bittner?

**Comment by bittner** [comment]
Django 2.0 needs to be added to [.travis.yml](https://github.com/jcassee/django-analytical/blob/master/.travis.yml). (_env_ section, _matrix/exclude_ section) and [setup.py](https://github.com/jcassee/django-analytical/blob/master/setup.py#L81) (trove).

File: analytical/utils.py [hunk 1]
```diff
@@ -71,7 +83,7 @@ def get_identity(context, prefix=None, identity_func=None, user=None):
         try:
             if user is None:
                 user = get_user_from_context(context)
-            if user.is_authenticated():
+            if get_user_is_authenticated(user):
                 if identity_func is not None:
```
Comment (bittner) [inline analytical/utils.py]: I know this is not part of the changes of this PR, but ...

Simply `if identity_func:` would be more pythonic.

File: analytical/utils.py [hunk 2]
```diff
@@ -49,6 +49,18 @@ def get_user_from_context(context):
     return None
 
 
+def get_user_is_authenticated(user):
+    """Check if the user is authenticated.
+
+    This is a compatibility function needed to support both Django 1.x and 2.x;
+    Django 2.x turns the function into a proper boolean so function calls will
+    fail."""
```
Comment (bittner) [inline analytical/utils.py]: I would prefer to have the `"""` in a separate line, both for the start and the end of the docstring.

File: analytical/utils.py (hunk 1 above)
Comment (jcassee) [reply analytical/utils.py]: Are you sure? `is not None` is more explicit I think?

**Comment by jcassee** [comment]
Thanks, Matthäus. I'll release a new version with this change shortly.

**Comment by jcassee** [comment]
Released as 2.4.0.

**Pull request merged**